### PR TITLE
Relaxed inUse definition

### PR DESCRIPTION
**Pull request by infeo**
This PR also mitigates problems described in #79, but with a different approach. Instead of timing based staleness of opened files, we determine if the filesystem is "in use", if at least one open file exists, which was written to.

The idea behind this approach is to allow regular unmounting as long as no data loss happens on the filesystem. If an open file was written to, we require it to be properly closed by a `release()` call to let `isInUse()` return `false`.

### CODE REVIEW

File: src/main/java/org/cryptomator/frontend/fuse/OpenFile.java [hunk 1]
```diff
@@ -19,11 +19,15 @@ public class OpenFile implements Closeable {
 	private final Path path;
 	private final FileChannel channel;
 
+	//"volatile" is fine, since methods of OpenFile ar externally synchronized
```
Comment (overheadhunter) [inline src/main/java/org/cryptomator/frontend/fuse/OpenFile.java:22]:
```suggestion
	/**
	 * Whether any data has been changed on this file.
	 *
	 * This value only changes while holding a write lock, see {@link org.cryptomator.frontend.fuse.locks.LockManager}.
	 */
```

File: src/main/java/org/cryptomator/frontend/fuse/OpenFileFactory.java [hunk 2]
```diff
@@ -67,8 +67,15 @@ public void close(long fileHandle) throws ClosedChannelException, IOException {
...
-	public int getOpenFileCount(){
-		return openFiles.size();
+	/**
+	 * Tests, if there exists dirty {@link OpenFile}s
+	 * This method is neither atomic regarding the set of open files nor regarding each open file individually. Therefore, external synchronization is required.
+	 *
+	 * @return {@code true} if and only if at least one dirty {@link OpenFile} exists. Otherwise {@code false}.
+	 * @see OpenFile#isDirty()
+	 */
+	boolean hastDirtyOpenFiles() {
```
Comment (overheadhunter) [inline src/main/java/org/cryptomator/frontend/fuse/OpenFileFactory.java:77]: "has dirty files" sounds like adult content :sweat_smile:

```suggestion
	/**
	 * Tests, if any {@link OpenFile} is considered <em>dirty</em>, i.e. has pending changes.
	 * This method is neither atomic regarding the set of open files nor regarding each open file individually. Therefore, external synchronization is required.
	 *
	 * @return {@code true} if and only if at least one dirty {@link OpenFile} exists. Otherwise {@code false}.
	 * @see OpenFile#isDirty()
	 */
	boolean hasPendingChanges() {
```

**Review by overheadhunter — CHANGES_REQUESTED** [review]
looks good except for some minor naming/docs improvements

File: src/main/java/org/cryptomator/frontend/fuse/OpenFileFactory.java (hunk 2 above)
Comment (infeo) [reply src/main/java/org/cryptomator/frontend/fuse/OpenFileFactory.java]: Changed in https://github.com/cryptomator/fuse-nio-adapter/pull/80/commits/842f5076d475eef72082fca51bc6862286fa1819:

I didn't come up with a better name, but would appreaciate if we stick to "hasDirty...", since it does not suggest anything and simply reflects the code.

File: src/main/java/org/cryptomator/frontend/fuse/ReadOnlyAdapter.java [hunk 3]
```diff
@@ -320,7 +317,7 @@ public void close() throws IOException {
 	 * Attempts to get a specific error code that best describes the given exception.
 	 * As a side effect this logs the error.
 	 *
-	 * @param e An exception
+	 * @param e      An exception
```
Comment (overheadhunter) [inline src/main/java/org/cryptomator/frontend/fuse/ReadOnlyAdapter.java:320]: unrelated

File: src/main/java/org/cryptomator/frontend/fuse/ReadOnlyAdapter.java [hunk 4]
```diff
@@ -331,8 +328,8 @@ protected int getErrorCodeForGenericFileSystemException(FileSystemException e, S
 //			LOG.warn("{} {} failed, name too long.", opDesc);
 //			return -ErrorCodes.ENAMETOOLONG();
 //		} else {
-			LOG.error(opDesc + " failed.", e);
-			return -errno.eio();
+		LOG.error(opDesc + " failed.", e);
+		return -errno.eio();
```
Comment (overheadhunter) [inline src/main/java/org/cryptomator/frontend/fuse/ReadOnlyAdapter.java:332]: unrelated

**Review by overheadhunter — CHANGES_REQUESTED** [review]
Undo these two changes, the rest looks fine to me

File: src/main/java/org/cryptomator/frontend/fuse/ReadOnlyAdapter.java (hunk 4 above)
Comment (infeo) [reply src/main/java/org/cryptomator/frontend/fuse/ReadOnlyAdapter.java]: done in https://github.com/cryptomator/fuse-nio-adapter/pull/80/commits/a4e330bb0b8154847c9ee048f01b2788ad052a25

File: src/main/java/org/cryptomator/frontend/fuse/ReadOnlyAdapter.java (hunk 3 above)
Comment (infeo) [reply src/main/java/org/cryptomator/frontend/fuse/ReadOnlyAdapter.java]: done in https://github.com/cryptomator/fuse-nio-adapter/pull/80/commits/a4e330bb0b8154847c9ee048f01b2788ad052a25

**Review by overheadhunter — APPROVED** [review]
